### PR TITLE
Main concepts in Wechaty

### PR DESCRIPTION
**Pull request by nibble0101**
# Improve the Introduction and Explanation sections project

| Section | Sub-section | Technical Writer |
| --- | --- | --- |
| Introduction | Main concepts in Wechaty |  Chris Estepa, Joseph Mawa |

This  is **Main concepts in Wechaty** sub-section of the **Introduction** section

image
image
image

### CODE REVIEW

File: docusaurus/docs/main-concepts-in-wechaty.md [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+
+This section covers the main Wechaty concepts you need to be familiar with to understand the other sections of the documentation. The other sections of the documentation assume you are familiar with the following concepts:
+
+- **Bot**
+
+  A bot is a software application that is programmed to perform specific tasks. Bots usually imitate human behavior and perform certain repetitive tasks faster. Bots are automated therefore do not need to be started manually by a human being.
+
+- **Chatbot**
+
+  These are bots which are designed for interacting with humans in voice or text-based conversation. Chatbots simulate human conversation by responding to certain phrases or words with programmed responses. Wechaty is a software program for developing chatbots for Instant Messaging services such as Wechat and Whatsapp. You can check our [Introduction section](#placeholder-link) for a list of all the supported Instant Messaging systems.
```
Comment (Rohitesh-Kumar-Jain) [inline docusaurus/docs/main-concepts-in-wechaty.md]: not sure but to me ```that are``` seems better than ```which are```, what do you think?

**Review by Rohitesh-Kumar-Jain — APPROVED** [review]
Great work @nibble0101 and @chris-4444, I approve these changes.

File: docusaurus/docs/main-concepts-in-wechaty.md (hunk 1 above)
Comment (nibble0101) [reply docusaurus/docs/main-concepts-in-wechaty.md]: Thanks for the review. I have changed it to `that`.

File: docusaurus/docs/main-concepts-in-wechaty.md [hunk 2]
```diff
@@ -0,0 +1,39 @@
+---
+title: Main concepts in Wechaty
+---
+
+![Wechaty: Conversational RPA SDK for Chatbot Makers](/img/wechaty-logo.svg)
+
+This section covers the main Wechaty concepts you need to be familiar with to understand the other sections of the documentation. The other sections of the documentation assume you are familiar with the following concepts:
```
Comment (proudofsimin) [inline docusaurus/docs/main-concepts-in-wechaty.md]: This section explain the main concepts in Wechaty. If you are new to communication software and its applications (?), start from this glossary to familiarize yourself with the terminology.

**Review by proudofsimin — COMMENTED** [review]
I actually feel the word terminology or glossary might be better than "concepts". How do you think?

File: docusaurus/docs/main-concepts-in-wechaty.md (hunk 2 above)
Comment (nibble0101) [reply docusaurus/docs/main-concepts-in-wechaty.md]: Added suggested changes in the commit below.

**Review by proudofsimin — APPROVED** [review]
Thank you :)

**Comment by lijiarui** [comment]
hi @Rohitesh-Kumar-Jain As GSoD Project PR Review Rules in https://github.com/wechaty/wechaty.js.org/issues/1045

This PR should follow our rules and get enough approval, then I will review this PR. So I remove the `ready to merge` tag and feel free to add this tag when it gets enough approval.

**Comment by proudofsimin** [comment]
> hi @Rohitesh-Kumar-Jain As GSoD Project PR Review Rules in #1045
> 
> This PR should follow our rules and get enough approval, then I will review this PR. So I remove the `ready to merge` tag and feel free to add this tag when it gets enough approval.

Hi @chris-4444, Could you review and add the tag `ready to merge` afterwards? Thank you.

**Comment by Rohitesh-Kumar-Jain** [comment]
@chris-4444 I would like you to once go through the PR, and approve when you think it's ready to get merged.

**Comment by lijiarui** [comment]
> @chris-4444 I would like you to once go through the PR, and approve when you think it's ready to get merged.

Hi @chris-4444 , as @Rohitesh-Kumar-Jain said, I will remove the `ready to merge tag`, after it got enough approval, feel free to add it again.

**Review by chris-4444 — APPROVED** [review]
LGTM.

File: docusaurus/docs/main-concepts-in-wechaty.md [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+
+  These are bots that are designed for interacting with humans in voice or text-based conversation. Chatbots simulate human conversation by responding to certain phrases or words with programmed responses. Wechaty is a software program for developing chatbots for Instant Messaging services such as Wechat and Whatsapp. You can check our [Introduction section](#placeholder-link) for a list of all the supported Instant Messaging systems.
+
+- **Wechaty Puppet**
+
+  Wechaty Puppet is a standard for maintaining compatibility between Instant Messaging(IM) systems. Wechaty Puppet defines the universal Instant Messaging interface for compatibility between Wechaty API and variants of IM systems.
+
+- **Puppet provider**
+
+  Wechaty puppet provider refers to [RPA](#placeholder-link) module used for connecting Wechaty API to the corresponding Instant Messaging platform bot account. Each supported Instant Messaging system has its own Puppet provider. See [Puppet Providers](#placeholder-link) section for a complete list of puppet providers.
```
Comment (lijiarui) [inline docusaurus/docs/main-concepts-in-wechaty.md]: How about we change `Instant Messaging platform` to `Instant Messaging platform(such as WeChat, Whatsapp, and TikTok)`

We can also add some examples here:

For example, the Wechaty Puppet Provider who is supporting WeChat is named [wechaty-puppet-wechat](https://github.com/wechaty/wechaty-puppet-wechat), supporting Whatsapp is named [wechaty-puppet-whatsapp](https://github.com/wechaty/wechaty-puppet-whatsapp), and supporting Lark is named [wechaty-puppet-lark](https://github.com/wechaty/wechaty-puppet-lark).

File: docusaurus/docs/main-concepts-in-wechaty.md [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+
+- **Puppet service provider**
+
+  Puppet service providers refer to Wechaty developers who have developed puppets for the different Instant Messaging systems.
+
```
Comment (lijiarui) [inline docusaurus/docs/main-concepts-in-wechaty.md:30]: We should add `puppet service` concept here:

```
**puppet service**

Wechaty Puppet Service is gRPC for Wechaty Puppet Provider.

For example, we can cloudify the Wechaty Puppet Provider wechaty-puppet-padlocal to a Wechaty Puppet Service by running our Wechaty Puppet Service Token Gateway.
```

File: docusaurus/docs/main-concepts-in-wechaty.md [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+  Wechaty Puppet is a standard for maintaining compatibility between Instant Messaging(IM) systems. Wechaty Puppet defines the universal Instant Messaging interface for compatibility between Wechaty API and variants of IM systems.
+
+- **Puppet provider**
+
+  Wechaty puppet provider refers to [RPA](#placeholder-link) module used for connecting Wechaty API to the corresponding Instant Messaging platform bot account. Each supported Instant Messaging system has its own Puppet provider. See [Puppet Providers](#placeholder-link) section for a complete list of puppet providers.
+
+- **Puppet service provider**
+
+  Puppet service providers refer to Wechaty developers who have developed puppets for the different Instant Messaging systems.
```
Comment (lijiarui) [inline docusaurus/docs/main-concepts-in-wechaty.md]: Maybe we should change to `  Puppet service providers refer to Wechaty developers who provide puppets cloud service for the different Instant Messaging systems.`

@huan I'm not very sure about this, how do you think of this?

File: docusaurus/docs/main-concepts-in-wechaty.md [hunk 6]
```diff
@@ -0,0 +1,39 @@
...
+
+  A bot is a software application that is programmed to perform specific tasks. Bots usually imitate human behavior and perform certain repetitive tasks faster. Bots are automated therefore, do not need to be started manually by a human being.
+
+- **Chatbot**
+
+  These are bots that are designed for interacting with humans in voice or text-based conversation. Chatbots simulate human conversation by responding to certain phrases or words with programmed responses. Wechaty is a software program for developing chatbots for Instant Messaging services such as Wechat and Whatsapp. You can check our [Introduction section](#placeholder-link) for a list of all the supported Instant Messaging systems.
+
+- **Wechaty Puppet**
+
+  Wechaty Puppet is a standard for maintaining compatibility between Instant Messaging(IM) systems. Wechaty Puppet defines the universal Instant Messaging interface for compatibility between Wechaty API and variants of IM systems.
```
Comment (lijiarui) [inline docusaurus/docs/main-concepts-in-wechaty.md:17]: @huan It seems @nibble0101 combines the two concepts `Wechaty Puppet` and `Wechaty Puppet Abstract` as you mentioned in [Introducing Wechaty Puppet Service (Providers)](https://wechaty.js.org/2021/01/14/wechaty-puppet-service/). I feel combine them makes it more clear to understand, and I support combining them together.

What's your suggestion? Do you think we should remove the `wechaty puppet abstract` concept in the blog or add this concept here?

File: docusaurus/docs/main-concepts-in-wechaty.md [hunk 7]
```diff
@@ -0,0 +1,39 @@
...
+
+- **Puppet provider**
+
+  Wechaty puppet provider refers to [RPA](#placeholder-link) module used for connecting Wechaty API to the corresponding Instant Messaging platform bot account. Each supported Instant Messaging system has its own Puppet provider. See [Puppet Providers](#placeholder-link) section for a complete list of puppet providers.
+
+- **Puppet service provider**
+
+  Puppet service providers refer to Wechaty developers who have developed puppets for the different Instant Messaging systems.
+
+- **Wechaty service token**
```
Comment (lijiarui) [inline docusaurus/docs/main-concepts-in-wechaty.md]: Please change this to `Wechaty Puppet Service Token`

File: docusaurus/docs/main-concepts-in-wechaty.md [hunk 8]
```diff
@@ -0,0 +1,39 @@
...
+
+  This is a unique key for authorizing Wechaty puppet service access. This token is issued by Puppet service providers.
```
Comment (lijiarui) [inline docusaurus/docs/main-concepts-in-wechaty.md]: I thinks it should change to 

```
This is a unique key for authorizing the service from Wechaty Puppet Service Provider access. This token is issued by Puppet service providers.

```

**Review by lijiarui — CHANGES_REQUESTED** [review]

File: docusaurus/docs/main-concepts-in-wechaty.md (hunk 3 above)
Comment (nibble0101) [reply docusaurus/docs/main-concepts-in-wechaty.md]: Added.

File: docusaurus/docs/main-concepts-in-wechaty.md (hunk 4 above)
Comment (nibble0101) [reply docusaurus/docs/main-concepts-in-wechaty.md:30]: Added

File: docusaurus/docs/main-concepts-in-wechaty.md (hunk 5 above)
Comment (nibble0101) [reply docusaurus/docs/main-concepts-in-wechaty.md]: Added

File: docusaurus/docs/main-concepts-in-wechaty.md (hunk 7 above)
Comment (nibble0101) [reply docusaurus/docs/main-concepts-in-wechaty.md]: Added

File: docusaurus/docs/main-concepts-in-wechaty.md (hunk 8 above)
Comment (nibble0101) [reply docusaurus/docs/main-concepts-in-wechaty.md]: Added

File: docusaurus/docs/main-concepts-in-wechaty.md [hunk 9]
```diff
@@ -0,0 +1,47 @@
+---
+title: Main concepts in Wechaty
+---
+
+![Wechaty: Conversational RPA SDK for Chatbot Makers](/img/wechaty-logo.svg)
```
Comment (lijiarui) [inline docusaurus/docs/main-concepts-in-wechaty.md]: I feel we may move this image or change a more related image here.

**Review by lijiarui — CHANGES_REQUESTED** [review]
Just one more tips then I will merge this PR!

File: docusaurus/docs/main-concepts-in-wechaty.md (hunk 9 above)
Comment (nibble0101) [reply docusaurus/docs/main-concepts-in-wechaty.md]: Removed the logo.

**Review by lijiarui — APPROVED** [review]
LGTM

**Pull request merged**